content/docs/zero/health-checks.mdx

```diff
@@ -0,0 +1,123 @@
+
+## Cluster version is not current {#cluster-version-current}
+
+This health makes sure the cluster replica is running the latest release. It runs daily.
+
+**What causes this error**: The replica is not running the latest version of Pomerium Zero.
+
+**Steps to resolve**: Update your Pomerium Zero cluster configuration to run the latest version of Pomerium. See the [Pomerium Zero Upgrading] page to see steps for your deployment environment.
+
+## Cluster using an in-memory storage backend {#persistent-backend}
+
+This health check throws an error if it detects a cluster isn't connected to a persistent storage backend. It runs every time the cluster configuration is updated.
+
+**What causes this error**:
+
+By default, a cluster uses an in-memory storage backend to synchronize replicas. This does not persist your configuration. For production environments, we recommend connecting each cluster instance to its own dedicated PostgreSQL database.
+
+**Steps to resolve**: See the [**Databroker Storage Connection String**](/docs/reference/databroker#databroker-storage-connection-string) reference page to learn how to connect to a database instance in Pomerium.
+
+## Incompatible configuration {#config-databroker-build}
+
+This health check looks for configuration incompatibilities in a replica. It runs every time the cluster configuration is updated.
+
+**What causes this error**: This is caused by a version incompatibility in the cluster configuration. For example, you somehow apply a configuration change that isn't compatible with your version of Pomerium Zero.
+
+**Steps to resolve**: If you see this warning, please [contact support](https://discuss.pomerium.com/).
+
+## Cluster can't communicate with the storage backend {#storage-backend}
+
+This health check monitors the connection to a PostgreSQL database. It runs every time the cluster configuration is updated.
+
+**What causes this error**: Pomerium can't detect the connection to your PostgreSQL database.
+
+**Steps to resolve**: Check the parameters of your [**Databroker Storage Connection String**](/docs/reference/databroker#databroker-storage-connection-string).
+
+## There's an issue with cluster configuration {#xds-cluster}
+
+This health check tells you if Envoy accepted the cluster configuration or not after a configuration update.
+
+**What causes this error**: If you see this warning, please [contact support](https://discuss.pomerium.com/).
+
+## There's an issue with the network listener {#xds-listener}
+
+This health check makes sure the network listener is configured correctly. It runs every time the cluster configuration is updated.
+
+**What causes this error**: Typically, this error is caused when Pomerium's network listener can't bind to the cluster's port.
+
+**Steps to resolve**: If this is a port binding error, you may need to kill a process running on that port or specify a port to bind to.
```

Comment:
another frequent issue is lacking permissions binding to port. user may need to add `CAP_NET_BIND_SERVICE` capability to the process.